network/discovery.go

```diff
@@ -131,21 +128,43 @@ func (d *discovery) setup() error {
 
 	go d.run()
 
+	go d.setupTable()
+
+	return nil
+}
+
+func (d *discovery) addToTable(node *peer.AddrInfo) error {
+	// before we include peers on the routing table -> dial queue
+	// we have to add them to the peerstore so that they are
+	// available to all the libp2p services
+	d.srv.host.Peerstore().AddAddr(node.ID, node.Addrs[0], peerstore.AddressTTL)
+	if _, err := d.routingTable.TryAddPeer(node.ID, false, false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *discovery) setupTable() error {
+	// Sleep until network setup is done
+	time.Sleep(5 * time.Second)
```

Comment:
Why `5s`?

Reply:
Originally filling table was done 5 sec later after the network layer started. When I change this part of code, I noticed syncer couldn't connect because libp2p of peer was not started.

Reply:
@Kourin1996 is there a way to signal from the networking layer when it has started?

This looks very error prone, or in the best case scenario, we would delay the boot of the node by 5s always.

Reply:
Actually, this error doesn't occur inside the node. If You start validator and non-validators at the same time, non-validators try to connect to validator nodes for syncer. But validator's syncer service didn't begin at that time. That's why non-validator couldn't sync blocks

Reply:
Let me think another way to fix this issue

Reply:
@lazartravica @brkomir @zivkovicmilos I divided libp2p initialization into two function and make it start connecting after all gRPC server are ready